javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/collector/MuzzleCodeGenerator.java

```diff
@@ -445,38 +444,43 @@ private void generateMuzzleReferencesMethod(ReferenceCollector collector) {
 
             mv.visitLdcInsn(method.getName());
 
-            { // return type
-              mv.visitLdcInsn(method.getReturnType().getDescriptor());
+            {
+              // method return and parameter types must be passed by Type.getType() call to be
+              // properly shaded
```

Comment:
I don't understand this comment

Reply:
Oh, I tried just passing the method descriptor string, but then it turned out that the shading plugin does not treat it as package/class name - so I had to stay with `Type`s. I'll reword that comment.